src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/DeploymentProperties.cs

```diff
@@ -29,7 +29,7 @@ public partial class DeploymentProperties
         /// <summary>
         /// Initializes a new instance of the DeploymentProperties class.
         /// </summary>
-        public DeploymentProperties(object template = default(object), TemplateLink templateLink = default(TemplateLink), object parameters = default(object), ParametersLink parametersLink = default(ParametersLink), DeploymentMode? mode = default(DeploymentMode?), DebugSetting debugSetting = default(DebugSetting))
+        public DeploymentProperties(DeploymentMode? mode, object template = default(object), TemplateLink templateLink = default(TemplateLink), object parameters = default(object), ParametersLink parametersLink = default(ParametersLink), DebugSetting debugSetting = default(DebugSetting))
         {
```

Comment:
This is generated code. I'm not sure if there is a way to set a default value for enum in swagger. @amarzavery any idea about this?
Also @ravbhatnagar we set the default value to incremental in clients (PoSH/CLI). In SDK, we want users to specify the mode value so the validation fails on the client side, instead of sending the request and waiting for it to fail on the server side